src/software/thunderscope/gl/layers/gl_ball_placement_layer.py

```diff
@@ -0,0 +1,149 @@
+                or new_placement_point != self.ball_placement_point
+            ):
+                self.ball_placement_point = new_placement_point
+
+                self.placement_tolerance_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.placement_tolerance_graphic.show()
+
+                self.placement_target_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.placement_target_graphic.set_radius(
+                    BALL_PLACEMENT_TOLERANCE_RADIUS_METERS
+                )
+                self.placement_target_graphic.show()
+
+                self.robot_avoid_circle_graphic.set_position(
+                    self.ball_placement_point.x_meters,
+                    self.ball_placement_point.y_meters,
+                )
+                self.robot_avoid_circle_graphic.show()
+
+                self.ball_placement_countdown_graphic = GLTextItem(
+                    parentItem=self,
+                    font=QtGui.QFont("Roboto", 7, weight=700),
+                    color=Colors.RED_TEXT_COLOR,
+                )
+                self.ball_placement_countdown_graphic.setData(
+                    text=f"{BALL_PLACEMENT_TIME_LIMIT_S}s",
+                    pos=[
+                        self.ball_placement_point.x_meters
+                        + BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS / 2,
+                        self.ball_placement_point.y_meters
+                        + BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS
+                        + 0.1,
+                        0,
+                    ],
+                )
+                self.ball_placement_countdown_graphic.show()
+
+                self.placement_start_time = time.time()
+                self.ball_placement_point_hidden = False
+
+            if self.shrink_target:
+                self.placement_target_graphic.set_radius(
+                    self.placement_target_graphic.radius - 0.01
+                )
+                self.shrink_target = self.placement_target_graphic.radius >= 0
+            else:
+                self.placement_target_graphic.set_radius(
+                    self.placement_target_graphic.radius + 0.01
+                )
+                self.shrink_target = (
+                    self.placement_target_graphic.radius
+                    >= BALL_PLACEMENT_TOLERANCE_RADIUS_METERS
+                )
+
+            time_since_start = int(time.time() - self.placement_start_time)
+            if time_since_start <= BALL_PLACEMENT_TIME_LIMIT_S:
+                self.ball_placement_countdown_graphic.setData(
+                    text=f"{BALL_PLACEMENT_TIME_LIMIT_S - time_since_start}s"
+                )
+            else:
+                self.ball_placement_countdown_graphic.setData(text=f"{0}s")
```

Comment:
In the future we should update this to use the time remaining in the `Referee` proto (I think it has a value for the remaining time in there)